### PR TITLE
workflows: add boot testing to CI builds

### PR DESCRIPTION
**Pull request by mwasilew**

### CODE REVIEW

File: .github/workflows/build-yocto.yml [hunk 1]
```diff
@@ -115,3 +115,35 @@ jobs:
           (( retries == i )) && { echo 'Failed to publish artifacts'; exit 1; }
 
           echo Image available at: ${url}
+  boot-test:
```
Comment (ndechesne) [inline .github/workflows/build-yocto.yml:118]: Your commit says that you are adding to nightly builds, but you are effectively adding it to each action (merge to main, PR, and nightly).

File: .github/workflows/build-yocto.yml [hunk 2]
```diff
@@ -115,3 +115,35 @@ jobs:
...
+    strategy:
```
Comment (ndechesne) [inline .github/workflows/build-yocto.yml:119]: probably want to add 
needs:  compile

File: .github/workflows/build-yocto.yml [hunk 3]
```diff
@@ -115,3 +115,35 @@ jobs:
...
+      fail-fast: true
```
Comment (ndechesne) [inline .github/workflows/build-yocto.yml:120]: add:
 if: github.repository == 'quic-yocto/meta-qcom-hwe'

File: ci/lava/qcs6490-rb3gen2-core-kit.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        - export IMAGE_PATH=$PWD
+        - echo "DEVICE_TYPE=qcs6490-rb3gen2" >> $IMAGE_PATH/flash.settings
+        - cat $IMAGE_PATH/flash.settings
+    timeout:
+      minutes: 10
+    to: downloads
+- deploy:
+    images:
+      image:
+        url: downloads://core-image-minimal-{{DEVICE_TYPE}}.rootfs.qcomflash.tar.gz
```
Comment (ndechesne) [inline ci/lava/qcs6490-rb3gen2-core-kit.yml:21]: FWIW. we have a PR to switch to core-base-image , which should be merged soon.

File: .github/workflows/build-yocto.yml (hunk 1 above)
Comment (mwasilew) [reply .github/workflows/build-yocto.yml:118]: I don't know if it should run on all builds. I can change either way.

File: ci/lava/qcs6490-rb3gen2-core-kit.yml (hunk 4 above)
Comment (mwasilew) [reply ci/lava/qcs6490-rb3gen2-core-kit.yml:21]: this is a limitation in LAVA. I can't pass a variable here. Maybe setting a file name in workflow is a better idea.

File: .github/workflows/build-yocto.yml (hunk 3 above)
Comment (mwasilew) [reply .github/workflows/build-yocto.yml:122]: fixed.

File: .github/workflows/build-yocto.yml (hunk 2 above)
Comment (mwasilew) [reply .github/workflows/build-yocto.yml:121]: fixed.

File: .github/workflows/build-yocto.yml (hunk 1 above)
Comment (ricardosalveti) [reply .github/workflows/build-yocto.yml:118]: We should probably run on all builds for now, if it ends up being too much we can split things later.

Comment (ndechesne) [reply .github/workflows/build-yocto.yml:118]: lol. ricardo replied 1 minute faster than me ;)

Comment (mwasilew) [reply .github/workflows/build-yocto.yml:118]: yes, and we have a draw 1:1 :)

Comment (ricardosalveti) [reply .github/workflows/build-yocto.yml:118]: @mwasilew you decide :D

Comment (mwasilew) [reply .github/workflows/build-yocto.yml:118]: I think I have to limit the boot test to only merges and nightly builds. By default PRs don't have access to org/repository secrets.

File: ci/lava/qcs6490-rb3gen2-core-kit.yml (hunk 4 above)
Comment (mwasilew) [reply ci/lava/qcs6490-rb3gen2-core-kit.yml]: fixed

File: .github/workflows/build-yocto.yml (hunk 1 above)
Comment (mwasilew) [reply .github/workflows/build-yocto.yml:118]: I added new condition. When build is triggered by PR, the boot test step is skipped.

**Review by ricardosalveti — APPROVED** [review]

**Comment by mwasilew** [comment]
I have a working version now. Example workflow here: https://github.com/quic-yocto/meta-qcom-hwe/actions/runs/12140562626/job/33851381293
The build still doesn't boot, that's why the boot test job fails.

I'll delete the `nightly-boot-test` branch once this PR is merged.

**Review by quic-vkraleti — APPROVED** [review]

**Comment by mwasilew** [comment]
Thanks for merging this PR. 
Right now there is just one board in the LAB that works. I'm trying to fix the problem, but it may take some time.

**Comment by ndechesne** [comment]
hmm. why did we merge a workflow which we know does not work (yet)? that seems counter intuitive to me.

**Comment by mwasilew** [comment]
@ndechesne why do you say it doesn't work?

**Comment by ndechesne** [comment]
I noticed the last push to main failed, but the failure is the compile job, and got confused. However I am not able to find a boot-test that completed successfully. Do you expect the boot-test to pass right now?

**Comment by mwasilew** [comment]
Yes, the problem was that 2 devices in the lab are effectively "bricked". I disabled them until there is a solution. Here is example boot test that worked: https://lava.infra.foundries.io/scheduler/job/56965.

**Pull request merged**